File: Old/Removed 2026/2026-07-31-old GraphicsPath/Api/Region.cs
#pragma warning disable
using Alternet.Drawing;
using System;

namespace NativeApi.Api
{
    //https://docs.wxwidgets.org/3.2/classwx_region.html#a1edc6768118cf02749b46774a0ca37f9
    public class Region
    {
        public void Clear() { }

        //Returns a value indicating whether the given point is contained within the region.
        public int ContainsPoint(PointI pt) => default;
        public int ContainsRect(RectI rect) => default;
        //Returns true if the region is empty, false otherwise
        public bool IsEmpty() => default;

        public bool IsOk() => default;

        public void InitializeWithRegion(Region region) => throw new Exception();

        public void InitializeWithRect(RectI rect) => throw new Exception();

        public unsafe void InitializeWithPolygon(
            PointD* points, int pointsCount,
            FillMode fillMode = FillMode.Alternate,
            float scaleFactor = 1.0f) => throw new Exception();

        public void IntersectWithRect(RectI rect) => throw new Exception();

        public void IntersectWithRegion(Region region) => throw new Exception();

        public void UnionWithRect(RectI rect) => throw new Exception();

        public void UnionWithRegion(Region region) => throw new Exception();

        public void XorWithRect(RectI rect) => throw new Exception();

        public void XorWithRegion(Region region) => throw new Exception();

        public void SubtractRect(RectI rect) => throw new Exception();

        public void SubtractRegion(Region region) => throw new Exception();

        public void Translate(int dx, int dy) => throw new Exception();

        public RectI GetBounds() => throw new Exception();

        public bool IsEqualTo(Region other) => throw new Exception();

        public int GetRegionHashCode() => throw new Exception();
    }
}

File: Old/Removed 2026/2026-07-31-old GraphicsPath/Region.cpp
#include "Region.h"
#include "GraphicsPath.h"
#include "Control.h"

namespace Alternet::UI
{
    Region::Region()
    {
    }

    Region::~Region()
    {
    }

    void Region::InitializeWithRegion(Region* region)
    {
        _region = wxRegion(region->_region);
    }

    void Region::InitializeWithRect(const RectI& rect)
    {
        _region = wxRegion(rect);
    }

    void Region::Clear()
    {
        _region.Clear();
    }

    int Region::ContainsPoint(const PointI& pt)
    {
        return _region.Contains(pt);
    }

    int Region::ContainsRect(const RectI& rect)
    {
        return _region.Contains(rect);
    }

    bool Region::IsEmpty()
    {
        return _region.IsEmpty();
    }

    bool Region::IsOk()
    {
        return _region.IsOk();
    }

    void Region::InitializeWithPolygon(Point* points, int pointsCount, FillMode fillMode, float scaleFactor)
    {
        std::vector<wxPoint> wxPoints;
        wxPoints.reserve(pointsCount);
        for (int i = 0; i < pointsCount; i++)
        {
            wxPoints.push_back(fromDipSf(points[i], scaleFactor));
        }

        _region = wxRegion(pointsCount, &wxPoints[0], GraphicsPath::GetWxFillMode(fillMode));
    }

    wxRegion Region::GetRegion()
    {
        return _region;
    }

    wxWindow* Region::GetWindow()
    {
        return ParkingWindow::GetWindow();
    }

    void Region::IntersectWithRect(const RectI& rect)
    {
        _region.Intersect(rect);
    }

    void Region::IntersectWithRegion(Region* region)
    {
        _region.Intersect(region->GetRegion());
    }

    void Region::UnionWithRect(const RectI& rect)
    {
        _region.Union(rect);
    }

    void Region::UnionWithRegion(Region* region)
    {
        _region.Union(region->GetRegion());
    }

    void Region::XorWithRect(const RectI& rect)
    {
        _region.Xor(rect);
    }

    void Region::XorWithRegion(Region* region)
    {
        _region.Xor(region->GetRegion());
    }

    void Region::SubtractRect(const RectI& rect)
    {
        _region.Subtract(rect);
    }

    void Region::SubtractRegion(Region* region)
    {
        _region.Subtract(region->GetRegion());
    }

    void Region::Translate(int dx, int dy)
    {
        _region.Offset(dx, dy);
    }

    RectI Region::GetBounds()
    {
        return _region.GetBox();
    }

    bool Region::IsEqualTo(Region* other)
    {
        return _region.IsEqual(other->GetRegion());
    }

    int Region::GetRegionHashCode()
    {
        return (int)(int64_t)_region.GetRefData();
    }
}